### DISCUSSION TITLE
Js Webapps, cppagent and CORS

### DISCUSSION THREAD

**Discussion by robot-ranger** [mtconnect/cppagent#337] [Q&A] [answered]
Hi all,

Has anyone worked with cppagent in the context of a js web app? Someone has to have made a web app for mtconnect? 

How can we work with CORS? What is the default CORS policy in cppagent? Is it configurable?

**Comment by wsobel** [comment]
The demo is a web app example that animates a digital twin. CORS policies are configurable by adding custom HTTP headers. I’ll check there docs. (Sent from mobile)On Oct 11, 2023, at 01:21, robot-ranger ***@***.***> wrote:﻿
Hi all,
Has anyone worked with cppagent in the context of a js web app? Someone has to have made a web app for mtconnect?
How can we work with CORS? What is the default CORS policy in cppagent? Is it configurable?

**Reply by robot-ranger** [reply, marked as answer]
yea, my developer figured out the Http headers by reverse engineering the parsing code. we couldnt find it documented anywhere. 

if anyone else stumbles on this in the future, we added the following to agent.cfg to allow all cross-origin:

```
HttpHeaders {
    Access-Control-Allow-Origin = *
    Access-Control-Allow-Methods = GET
    Access-Control-Allow-Headers = Content-Type
}
```

I believe you could replace the * with a list of origins to specify whitelisted cross-origins

**Comment by wsobel** [comment]
Thanks, I’ll add this to the documentation. - WOn Oct 12, 2023, at 17:43, robot-ranger ***@***.***> wrote:﻿
yea, my developer figured out the Http headers by reverse engineering the parsing code. we couldnt find it documented anywhere.
if anyone else stumbles on this in the future, we added the following to agent.cfg to allow all cross-origin:
HttpHeaders {
    Access-Control-Allow-Origin = *
    Access-Control-Allow-Methods = GET
    Access-Control-Allow-Headers = Content-Type
}

I believe you could replace the * with a list of origins to specify whitelisted cross-origins

**Comment by wsobel** [comment]
Added to the README in the 2.2.0.10
